### PR TITLE
Test for DocList

### PR DESCRIPTION
**Pull request by tomijaga**
solves: #1348
account number: 8de30226230c35bbc1ce4a63c62a7b9c86bf0ce21fc7bc1a984b7884a9f88782

### CODE REVIEW

File: src/components/DocWrapper/DocList/DocList.test.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+describe('DocList', () => {
+  let props: DocListProps;
+
+  beforeEach(() => {
+    props = {
+      className: 'test-class',
+      variant: 'ul',
+    };
+  });
```
Comment (thesanjeevsharma) [inline src/components/DocWrapper/DocList/DocList.test.tsx:16]: What's the benefit of this? 🤔 If you need the same props every time, why use `beforeEach`?

File: src/components/DocWrapper/DocList/DocList.test.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/* eslint-disable react/jsx-props-no-spreading */
```
Comment (thesanjeevsharma) [inline src/components/DocWrapper/DocList/DocList.test.tsx:1]: You can remove this. Take a pull from development. Test files have been allowed props spreading.

**Review by thesanjeevsharma — CHANGES_REQUESTED** [review]

File: src/components/DocWrapper/DocList/DocList.test.tsx (hunk 1 above)
Comment (tomijaga) [reply src/components/DocWrapper/DocList/DocList.test.tsx]: I fixed it and just initialized the props variable with the values

File: src/components/DocWrapper/DocList/DocList.test.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import {render, screen} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import DocList, {DocListProps} from './index';
+
+describe('DocList', () => {
+    const props: DocListProps = {
+    className: 'test-class',
```
Comment (jamessspanggg) [inline src/components/DocWrapper/DocList/DocList.test.tsx:9]: The indentation here should be two spaces after `const`

File: src/components/DocWrapper/DocList/DocList.test.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    variant: 'ul',
+    };
+
+  it('renders without crashing', () => {
+    render(<DocList {...props}>Hello World</DocList>);
+  });
+
+  it('renders with proper default className', () => {
+    render(<DocList {...props}>Hello World</DocList>);
+    const list = screen.getByTestId(`DocList--${props.variant}`);
```
Comment (jamessspanggg) [inline src/components/DocWrapper/DocList/DocList.test.tsx:19]: I think you can create a `const testIdUl = 'DocList--ul` and `const testIdOl = 'DocList--ol` and reuse them all over

File: src/components/DocWrapper/DocList/DocList.test.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    expect(list).toHaveClass('DocList');
+  });
+
+  it('renders unorderd list when specified in variant prop', () => {
```
Comment (jamessspanggg) [inline src/components/DocWrapper/DocList/DocList.test.tsx:24]: *unordered

File: src/components/DocWrapper/DocList/DocList.test.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    render(<DocList {...props}>Hello World</DocList>);
+    const list = screen.getByTestId(`DocList--ul`);
+
+    expect(list).toBeTruthy();
+    expect(list).toHaveClass(`DocList--ul`);
+  });
+
+  it('renders unorderd list with passed in className', () => {
```
Comment (jamessspanggg) [inline src/components/DocWrapper/DocList/DocList.test.tsx:32]: here too

File: src/components/DocWrapper/DocList/DocList.test.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    render(<DocList {...props}>Hello World</DocList>);
+    const list = screen.getByTestId(`DocList--ul`);
+
+    expect(list).toHaveClass('test-class');
+  });
+
+  it('renders orderd list when specified in variant prop', () => {
```
Comment (jamessspanggg) [inline src/components/DocWrapper/DocList/DocList.test.tsx:39]: here too

File: src/components/DocWrapper/DocList/DocList.test.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    render(<DocList variant="ol">Hello World</DocList>);
+    const list = screen.getByTestId(`DocList--ol`);
+
+    expect(list).toBeTruthy();
+    expect(list).toHaveClass(`DocList--ol`);
+  });
+
+  it('renders orderd list with passed in className', () => {
```
Comment (jamessspanggg) [inline src/components/DocWrapper/DocList/DocList.test.tsx:47]: here

**Review by jamessspanggg — COMMENTED** [review]
@angle943 @thesanjeevsharma 

I have a suggestion. Should we create a data-testId variable in the original component file. i.e. `export const dataTestId = ...` and the test file can just import it. This will ensure re-usability and decrease the mistake of passing the `testId` to `getByTestId` wrongly

**Comment by angle943** [comment]
> @angle943 @thesanjeevsharma
> 
> I have a suggestion. Should we create a data-testId variable in the original component file. i.e. `export const dataTestId = ...` and the test file can just import it. This will ensure re-usability and decrease the mistake of passing the `testId` to `getByTestId` wrongly

@jamessspanggg hmm the only problem with that is, what if we have more than one data-testid in a component? Also, `data-testid`s should have same name as `classNames`, so why don't we export the className too?

But there might be some refactors we can do in the future regarding the fact that classNames and data-testid have parity... hmm let's think about this for a bit

File: src/components/DocWrapper/DocList/DocList.test.tsx (hunk 3 above)
Comment (angle943) [reply src/components/DocWrapper/DocList/DocList.test.tsx:9]: also please move the `const props` out of the describe please

**Review by angle943 — APPROVED** [review]

**Pull request merged**